content-security-policy/webrtc/webrtc.js

```diff
@@ -0,0 +1,62 @@
+
+// Creates two RTCPeerConnection and tries to connect them. Returns
+// "allowed" if the connection is permitted, "blocked" if it is
+// blocked on both sides and "inconsistent" in the even that the
+// result is the same on both sides (should never happen).
+async function tryConnect() {
+    const iceServers = [{urls: "stun:stun.l.google.com:19302"}];
+    const pc1 = new RTCPeerConnection({iceServers});
+    const pc2 = new RTCPeerConnection({iceServers});
+
+    // Returns a promise which resolves to a boolean which is true
+    // if and only if pc.iceConnectionState settles in the "failed"
+    // state, and never transitions to any state other than "new"
+    // or "failed."
+    const pcFailed = (pc) => {
+        pc.onicecandidate = ({candidate}) => {
+            pc.addIceCandidate(candidate);
```

Comment:
To be clear, the unhandled rejection is happening here.

Reply:
This is calling addIceCandidate on the wrong pc. pc1 should call it on pc2 and vice versa (this being a shortcut-ed signaling dance). See suggestion below.